src/async_wrap.cc

```diff
@@ -171,14 +168,8 @@ void AsyncWrap::EmitPromiseResolve(Environment* env, double async_id) {
 
   Local<Value> async_id_value = Number::New(env->isolate(), async_id);
   Local<Function> fn = env->async_hooks_promise_resolve_function();
-  TryCatch try_catch(env->isolate());
-  MaybeLocal<Value> ar = fn->Call(
-      env->context(), Undefined(env->isolate()), 1, &async_id_value);
-  if (ar.IsEmpty()) {
-    ClearFatalExceptionHandlers(env);
-    FatalException(env->isolate(), try_catch);
-    UNREACHABLE();
-  }
+  FatalTryCatch try_catch(env);
+  (void)fn->Call(env->context(), Undefined(env->isolate()), 1, &async_id_value);
```

Comment:
The `(void)...` is to suppress the `-Wunused-result` warning?  That works with clang but gcc still emits a warning, AFAIK.

Reply:
@bnoordhuis It appears this depends on the gcc version: gcc 6 doesn’t complain, gcc 7 does? I think older versions of it were also fine with it…

I’ve went with a dummy `.FromMaybe` call, that doesn’t seem quite as pretty but I’m not sure what a better way would be

Reply:
It used to be a long-standing gcc bug that was only partially fixed: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509 from 2005.

I suppose now might be a good a time as any to introduce a Use() or USE() function:
```c++
template <typename T> void USE(T&&) {}
```
Good idea, bad idea?  If bad idea, I'm fine with `(void)...`.

Reply:
It’s a good idea, I’ve added a new commit with that (both for the code bits here and those where we use `(void)`.